x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/action/jinaai/JinaAIActionCreator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.action.jinaai;
+
+import org.elasticsearch.inference.InputType;
+import org.elasticsearch.xpack.inference.external.action.ExecutableAction;
+import org.elasticsearch.xpack.inference.external.action.SenderExecutableAction;
+import org.elasticsearch.xpack.inference.external.http.sender.JinaAIEmbeddingsRequestManager;
+import org.elasticsearch.xpack.inference.external.http.sender.JinaAIRerankRequestManager;
+import org.elasticsearch.xpack.inference.external.http.sender.Sender;
+import org.elasticsearch.xpack.inference.services.ServiceComponents;
+import org.elasticsearch.xpack.inference.services.jinaai.embeddings.JinaAIEmbeddingsModel;
+import org.elasticsearch.xpack.inference.services.jinaai.rerank.JinaAIRerankModel;
+
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.inference.external.action.ActionUtils.constructFailedToSendRequestMessage;
+
+/**
+ * Provides a way to construct an {@link ExecutableAction} using the visitor pattern based on the jinaai model type.
+ */
+public class JinaAIActionCreator implements JinaAIActionVisitor {
+    private final Sender sender;
+    private final ServiceComponents serviceComponents;
+
+    public JinaAIActionCreator(Sender sender, ServiceComponents serviceComponents) {
+        // TODO Batching - accept a class that can handle batching
```

Comment:
Just for my understanding of this `TODO`: what do you mean with batching here? Batching a set of requests over some window (time, amount of requests etc.) into one request and sending them to your API once?

Reply:
to be honest, this remains here from the other providers where I took inspiration from.